### PR TITLE
Rotate operator update indices

### PR DESCRIPTION
**Pull request by ebatsell**
Rotates the operator index we start on based on epoch % num_operators. Internally just tracks a counter from 0..num_operators and increments that, converting to the expected index number in check_and_update_index().

### CODE REVIEW

**Review by coachchucksol — COMMENTED** [review]
What happens if operators are added/removed mid update loop

File: vault_core/src/vault_update_state_tracker.rs [hunk 1]
```diff
@@ -59,8 +59,35 @@ impl VaultUpdateStateTracker {
...
 
-    pub fn last_updated_index(&self) -> u64 {
-        self.last_updated_index.into()
+    pub fn update_counter(&self) -> u64 {
+        self.update_counter.into()
+    }
+
+    /// Calculates target index for the vault update state tracker
+    /// factoring in where we started, and accounting for wraparound
+    pub fn last_updated_index(&self, num_operators: u64) -> Result<u64, VaultError> {
```
[the remark this answers is not in the document]
Comment (ebatsell) [reply vault_core/src/vault_update_state_tracker.rs]: operators can't get removed, but adding would be an issue I think

[the remark this answers is not in the document]
Comment (ebatsell) [reply vault_core/src/vault_update_state_tracker.rs]: update: this is not an issue, operator adds are locked

**Review by coachchucksol — CHANGES_REQUESTED** [review]
Fix these issues

**Review by coachchucksol — APPROVED** [review]
lgtm

File: vault_core/src/vault_update_state_tracker.rs [hunk 2]
```diff
@@ -196,15 +230,50 @@ mod tests {
...
-            .check_and_update_index(1)
+            .check_and_update_index(1, n)
             .unwrap();
         assert_eq!(
-            vault_update_state_tracker.check_and_update_index(3),
+            vault_update_state_tracker.check_and_update_index(3, n),
+            Err(VaultError::VaultUpdateIncorrectIndex)
+        );
+    }
+
```
[the remark this answers is not in the document]
Comment (coachchucksol) [reply vault_core/src/vault_update_state_tracker.rs:246]: Note, this is done in the test `test_all_operators_updated` where the ncn epoch is 0 and n = 4. 0%4 = 0

**Review by buffalu — APPROVED** [review]

**Review by buffalu — APPROVED** [review]

File: integration_tests/tests/fixtures/vault_client.rs [hunk 3]
```diff
@@ -1051,16 +1051,20 @@ impl VaultProgramClient {
             .get_config(&Config::find_program_address(&jito_vault_program::id()).0)
             .await?;
 
+        let ncn_epoch = slot / config.epoch_length();
```
[the remark this answers is not in the document]
Comment (coachchucksol) [reply integration_tests/tests/fixtures/vault_client.rs:956]: It's in integration tests - do we still need check arithmetic?

[the remark this answers is not in the document]
Comment (buffalu) [reply integration_tests/tests/fixtures/vault_client.rs:956]: oh u right u right

[the remark this answers is not in the document]
Comment (buffalu) [reply integration_tests/tests/fixtures/vault_client.rs:956]: dont need to fix

**Pull request merged**